specification/cognitiveservices/data-plane/ContentSafety/preview/2023-04-30-preview/contentsafety.json

```diff
@@ -0,0 +1,1066 @@
+                "value"
+              ],
+              "x-typespec-name": "Azure.Core.Foundations.CustomPage<TextBlocklist, Azure.Core.(anonymous model)>"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "headers": {
+              "x-ms-error-code": {
+                "description": "String error code indicating what went wrong.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Azure.Core.Foundations.ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get All Text Blocklists": {
+            "$ref": "./examples/ListTextBlocklists.json"
+          }
+        }
+      }
+    },
+    "/text/blocklists/{blocklistName}": {
+      "get": {
+        "operationId": "TextBlocklists_GetTextBlocklist",
+        "summary": "Get Text Blocklist By blocklistName",
+        "description": "Returns text blocklist details.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "blocklistName",
+            "in": "path",
+            "required": true,
+            "description": "Text blocklist name. Only supports the following characters: 0-9  A-Z  a-z  -  .  _  ~",
+            "maxLength": 64,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/TextBlocklist"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "headers": {
+              "x-ms-error-code": {
+                "description": "String error code indicating what went wrong.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Azure.Core.Foundations.ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Text Blocklist By blocklistName": {
+            "$ref": "./examples/GetTextBlocklist.json"
+          }
+        }
+      },
+      "patch": {
+        "operationId": "TextBlocklists_CreateOrUpdateTextBlocklist",
+        "summary": "Create Or Update Text Blocklist",
+        "description": "Updates a text blocklist, if blocklistName does not exist, create a new blocklist.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "blocklistName",
+            "in": "path",
+            "required": true,
+            "description": "Text blocklist name. Only supports the following characters: 0-9  A-Z  a-z  -  .  _  ~",
```

Comment:
Rather than put allowed characters in the description, you should specify an explicit pattern.